src/renderers/shared/stack/reconciler/ReactRef.js

```diff
@@ -19,8 +19,30 @@ import type { ReactElement } from 'ReactElementType';
 
 var ReactRef = {};
 
+if (__DEV__) {
+  var ReactCompositeComponentTypes = require('ReactCompositeComponentTypes');
+  var ReactComponentTreeHook = require('ReactComponentTreeHook');
+  var warning = require('warning');
+}
+
 function attachRef(ref, component, owner) {
   if (typeof ref === 'function') {
+    if (__DEV__) {
+      let info = '';
+      const ownerName = owner && owner.getName();
+      if (ownerName) {
+        info += ' Check the render method of `' + ownerName + '`.';
+      }
+
+      warning(
+        /* $FlowFixMe component._compositeType really exists I swear. */
+        component._compositeType !== ReactCompositeComponentTypes.StatelessFunctional,
+        'Stateless function components cannot be given refs. ' +
+        'Attempts to access this ref will fail.%s%s',
+        info,
+        ReactComponentTreeHook.getStackAddendumByID((owner || component)._debugID)
```

Comment:
Similarly you want the `component`'s `debugID`, not the `owner`'s. I think `component` is guaranteed to exist since we're calling `component.getPublicInstance()` on the next line.